src/api/app/models/event/workflow_run_fail.rb

```diff
@@ -18,7 +18,12 @@ def token_executors
     end
 
     def token_members
-      [token&.users, token&.groups].flatten.compact
+      # TODO: make sure this cannot be done in `find_for_event.rb` instead
+
+      # exctract all the users and groups members the token is shared with,
+      # and merge them all together in a single set removing nils and duplicated entries
+      group_users = token&.groups&.map(&:users)&.flatten
+      [token&.users, group_users].flatten.compact.uniq
```

Comment:
Why would you do this here and not in `Token::Workflow` anyway?

Reply:
That is a good point :+1:

Reply:
Done